share/commands/agent-reset

```diff
@@ -81,18 +78,18 @@ if [ -f "${RUDDER_VAR}/cfengine-community/policy_server.dat" ]
 then
   if [ "${UUID}" != "root" ]
   then
-    # the update should do the rest
+    # Bootstrap promises: the update should do the rest
     cp /opt/rudder/share/bootstrap-promises/* /var/rudder/cfengine-community/inputs/
 
     # - update to make sure we have at least the initial promises
     [ "$VERBOSE" = true ] && echo "Getting initial promises..."
     # $@ to pass options through
     rudder agent update "$@"
   else
-    # Ensure that ncf is up to date  
-    [ "$VERBOSE" = true ] && echo "Ensure ncf is up to date..."
-    # $@ to pass options through
-    rudder agent update "$@"
+    # Initial promises
+    [ "$VERBOSE" = true ] && echo "Restoring initial promises..."
+    cp -r /opt/rudder/share/initial-promises/* /var/rudder/cfengine-community/inputs/
+    cp -r /usr/share/ncf/tree/* /var/rudder/ncf/common/
```

Comment:
why not rudder agent update instead of copying the folder with cp ?

Reply:
especially as you are removing /var/rudder/ncf/local

Reply:
because we need working promises to update

Reply:
you already copied initial promises, so update works